src/Avalonia.Controls/NumericUpDown/NumericUpDown.cs

```diff
@@ -485,6 +508,20 @@ protected virtual void OnTextChanged(string? oldValue, string? newValue)
                 SyncTextAndValueProperties(true, Text);
             }
         }
+        
+        /// <summary>
+        /// Called when the <see cref="Text"/> property value changed.
+        /// </summary>
+        /// <param name="oldValue">The old value.</param>
+        /// <param name="newValue">The new value.</param>
+        protected virtual void OnTextConverterChanged(IValueConverter? oldValue, IValueConverter? newValue)
+        { 
+            if (!_internalValueSet && IsInitialized)
+            {
+                SyncTextAndValueProperties(false, null, true);
+            }
+            SetValidSpinDirection();
```

Comment:
I believe `SetValidSpinDirection` is only dependent on `Value`, so it's not required when text representation format is changed. So `OnTextConverterChanged` should be the same as `OnFormatStringChanged`: 
https://github.com/AvaloniaUI/Avalonia/blob/215e8897c2c7cc797a6e82d7a8df1612263261d2/src/Avalonia.Controls/NumericUpDown/NumericUpDown.cs#L434-L440

Reply:
Alright, done.